src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -147,36 +147,48 @@ static string PokeTheRegistryForLocalClonePath()
         const string MRUKeyPath = "MRUSettingsLocalProjectLocationEntries";
         public string SetDefaultProjectPath(string path)
         {
-            string old;
-            using (var newProjectKey = Registry.CurrentUser.OpenSubKey(NewProjectDialogKeyPath, true))
+            var old = String.Empty;
+            try
             {
-                using (var mruKey = newProjectKey?.OpenSubKey(MRUKeyPath, true))
+                var newProjectKey = Registry.CurrentUser.OpenSubKey(NewProjectDialogKeyPath, true);
+                if (newProjectKey == null)
+                    newProjectKey = Registry.CurrentUser.CreateSubKey(NewProjectDialogKeyPath);
+
+                using (newProjectKey)
                 {
+                    var mruKey = newProjectKey.OpenSubKey(MRUKeyPath, true);
                     if (mruKey == null)
-                        return String.Empty;
-
-                    // is this already the default path? bail
-                    old = (string)mruKey.GetValue("Value0", string.Empty, RegistryValueOptions.DoNotExpandEnvironmentNames);
-                    if (String.Equals(path.TrimEnd('\\'), old.TrimEnd('\\'), StringComparison.CurrentCultureIgnoreCase))
-                        return old;
+                        mruKey = Registry.CurrentUser.CreateSubKey(MRUKeyPath);
 
-                    // grab the existing list of recent paths, throwing away the last one
-                    var numEntries = (int)mruKey.GetValue("MaximumEntries", 5);
-                    var entries = new List<string>(numEntries);
-                    for (int i = 0; i < numEntries - 1; i++)
+                    using (mruKey)
                     {
-                        var val = (string)mruKey.GetValue("Value" + i, String.Empty, RegistryValueOptions.DoNotExpandEnvironmentNames);
-                        if (!String.IsNullOrEmpty(val))
-                            entries.Add(val);
-                    }
+                        // is this already the default path? bail
+                        old = (string)mruKey.GetValue("Value0", string.Empty, RegistryValueOptions.DoNotExpandEnvironmentNames);
```

Comment:
Can't be null, it's locked in the using statement, and same logic as the newProjectKey, CreateSubKey will either return something or throw, in which case it will land on the exception handler.